salt/states/user.py

```diff
@@ -854,8 +847,13 @@ def _change_homedir(name, val):
                 "logonscript": win_logonscript,
             }
 
-        if __salt__["user.add"](**params):
-            ret["comment"] = "New user {0} created".format(name)
+        result = __salt__["user.add"](**params)
+        if result is not True and salt.utils.platform.is_windows():
```

Comment:
Why is this Windows specific?
If it is Windows specific, what happens when it is `not True` but not Windows?